### DISCUSSION TITLE
Renaming and deleting on crates.io

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [rust-lang/crates.io#9661] [Feature Requests]
**You may want to delete a crate you uploaded for several reasons:**

1. You accidentally uploaded the wrong crate
2. You were testing or playing with uploading to crates.io
3. You changed your mind about publishing the crate
4. Various other reasons...

If the uploaded crate has no dependants, then why not allow it to be renamed or deleted?

**Use case:**

I uploaded a test crate just to see how crates.io worked, foolishly thinking that "of course I will be able to delete it". Now the crate is stuck there forever.

**Comment by eth3lbert** [comment]
Please see https://github.com/rust-lang/rfcs/pull/3660 for details on crate deletion. You can track the progress through https://github.com/rust-lang/crates.io/issues/9352.
